src/dune_rules/lib_file_deps.ml

```diff
@@ -7,23 +7,25 @@ module Group = struct
 
   type t =
     | Ocaml of ocaml
-    | Melange of Melange.Cm_kind.t
+    | Melange of Melange.Lib_file_deps_group.t
     | Header
 
-  let all = [ Ocaml Cmi; Ocaml Cmx; Melange Cmi; Melange Cmj; Header ]
+  let all =
+    [ Ocaml Cmi; Ocaml Cmx; Melange Cmi; Melange Cmj; Melange Js; Header ]
 
   let ext = function
     | Ocaml Cmi -> Cm_kind.ext Cmi
     | Ocaml Cmx -> Cm_kind.ext Cmx
     | Melange Cmi -> Lib_mode.Cm_kind.ext (Melange Cmi)
     | Melange Cmj -> Lib_mode.Cm_kind.ext (Melange Cmj)
+    | Melange Js -> Melange.js_ext
     | Header -> Foreign_language.header_extension
 
   let obj_dir t obj_dir =
     match t with
     | Ocaml Cmi -> Obj_dir.public_cmi_dir obj_dir
     | Ocaml Cmx -> Obj_dir.native_dir obj_dir
-    | Melange (Cmi | Cmj) -> Obj_dir.melange_dir obj_dir
+    | Melange (Cmi | Cmj | Js) -> Obj_dir.melange_dir obj_dir
```

Comment:
We don't produce Js in the obj_dir so this seems suspicious

Reply:
Rather, I think we are producing into the obj_dir, but it seems easier  to avoid that and just produce into melange.emit directory.

Reply:
You are right, it was a stretch to add `Melange Js` to `Lib_file_deps.Group`. The obj_dir is not necessary at all for .js artifacts.

I removed this group and used a glob instead in https://github.com/ocaml/dune/pull/6349/commits/86e4deb66f69e3689b2b0e5a29ae6a60453090ef.